src/construction/construction.py

```diff
@@ -1,282 +1,81 @@
-        return additions_df
-    else:
-        construction_logger.info("No additions found.")
-        return None
+        return snapshot_df
 
-
-def output_construction_files(amendments_df, additions_df, config, write_csv, run_id):
-    """Save CSVs of amendments and additions for user approval."""
-    # Prepare output paths
-    network_or_hdfs = config["global"]["network_or_hdfs"]
-    paths = config[f"{network_or_hdfs}_paths"]
-    tdate = datetime.now().strftime("%Y-%m-%d")
-    construction_folder = paths["construction_path"]
-    amendments_filename = os.path.join(
-        construction_folder, "auto_construction", f"construction_amendments_{tdate}_v{run_id}.csv"
-    )
-    additions_filename = os.path.join(
-        construction_folder, "auto_construction", f"construction_additions_{tdate}_v{run_id}.csv"
-    )
-
-    # Check if the dataframes are empty before writing
-    if amendments_df is not None:
-        write_csv(f"{amendments_filename}", amendments_df)
-    if additions_df is not None:
-        write_csv(f"{additions_filename}", additions_df)
-
-    return True
-
-
-def apply_construction(main_df, config, check_file_exists, read_csv, write_csv, run_id):
-    """Read user-edited construction files and apply them to the main snapshot."""
-    # Prepare filepaths to read from
-    network_or_hdfs = config["global"]["network_or_hdfs"]
-    paths = config[f"{network_or_hdfs}_paths"]
-    amendments_filepath = paths["construction_amend_path"]
-    additions_filepath = paths["construction_add_path"]
-
-    # Check if the construction files exist
-    amendments_exist = check_file_exists(amendments_filepath)
-    additions_exist = check_file_exists(additions_filepath)
-
-    # If each file exists, read it and call the function to apply them
-    if (not amendments_exist) and (not additions_exist):
-        construction_logger.info("No amendments or additions to apply, skipping...")
-        return main_df
-
-    if amendments_exist:
-        try:
-            amendments_df = read_csv(amendments_filepath)
-            constructed_df = apply_amendments(main_df, amendments_df)
-        except pd.errors.EmptyDataError:
-            construction_logger.warning(
-                f"Amendments file {amendments_filepath} is empty, skipping..."
-            )
-
-    if additions_exist:
+    # Check the construction file exists and has records, then read it
+    construction_file_path = config["network_paths"]["construction_file_path"]
+    construction_file_exists = check_file_exists(construction_file_path)
+    if construction_file_exists:
         try:
-            additions_df = read_csv(additions_filepath)
-            additions_df["instance"] = additions_df["instance"].astype("Int64")
-            constructed_df = apply_additions(main_df, additions_df)
+            construction_df = read_csv(construction_file_path)
         except pd.errors.EmptyDataError:
             construction_logger.warning(
```

Comment:
I like this, I hadn't seen it before. 
Could we add it to the read_csv func in hdfs_mods and local_mods?